libc/unistd_fsdir.c

```diff
@@ -72,44 +76,57 @@ TEST(unistd_fsdir, getcwd)
 
 	TEST_ASSERT_NULL(getcwd(buf, 1));
 	TEST_ASSERT_EQUAL_INT(ERANGE, errno);
+
+	/* go back to the test working directory */
+	TEST_ASSERT_EQUAL_INT(0, chdir(twd));
 }
 
 
 TEST(unistd_fsdir, chdir_abs)
 {
-	/* test chdir to some directory */
-	TEST_ASSERT_EQUAL_INT(0, chdir("/dev"));
-	TEST_ASSERT_NOT_NULL(getcwd(buf, sizeof(buf)));
-	TEST_ASSERT_EQUAL_STRING("/dev", buf);
-
-	/* test chdir to cwd */
-	TEST_ASSERT_EQUAL_INT(0, chdir("/dev"));
-	TEST_ASSERT_NOT_NULL(getcwd(buf, sizeof(buf)));
-	TEST_ASSERT_EQUAL_STRING("/dev", buf);
-
-	/* test chdir back to root */
+	/* test chdir to root */
 	TEST_ASSERT_EQUAL_INT(0, chdir("/"));
 	TEST_ASSERT_NOT_NULL(getcwd(buf, sizeof(buf)));
 	TEST_ASSERT_EQUAL_STRING("/", buf);
+
+	/* test chdir back to test working directory */
+	TEST_ASSERT_EQUAL_INT(0, chdir(twd));
+	TEST_ASSERT_NOT_NULL(getcwd(buf, sizeof(buf)));
+	TEST_ASSERT_EQUAL_STRING(twd, buf);
 }
 
 
 TEST(unistd_fsdir, chdir_rel)
 {
+	char absPath[PATH_MAX];
+	int slen;
+
+	TEST_ASSERT_NOT_NULL(strcpy(absPath, twd));
+
+	slen = strlen(absPath);
+	if (absPath[slen - 1] != '/') {
+		TEST_ASSERT_NOT_NULL(strcat(absPath, "/"));
```

Comment:
don't ever use `strcat` without checking if the result would fit in the destination buffer (possible buffer overflow), use `strncat` or `strncpy`